### ISSUE TITLE
Problem with marker designing in GMATA

### ISSUE THREAD

**Issue by NHKisr** [XuewenWangUGA/GMATA#3]
I successfully installed the GMATA software on my MAC and was able to run it upto the Gbrowse viewing. While performing the marker designing I was stuck. The program started running and four output files were generated. However, the .mk file has nothing but headings. I thought something is wrong with my files. So I tried the example file (testseqfasta) given in the GMATA folder. But the same problem arises with this file. I don't know whats wrong. Please advise.

**Comment by XuewenWangUGA** [comment]
It seems that your installation is not completed. Please refer to manual or information at https://sourceforge.net/p/gmata/discussion/general/ 
You have to install two dependents ePCR and primer3  in Mac OS. For Windows system, no dependents are required because I already built everything.
The tested working dependents are available for downloading at https://sourceforge.net/projects/gmata/files/

**Comment by NHKisr** [comment]
thanks for your response!
I have downloaded the required dependencies but am confused as to how to go about installing them. I am not able to follow the instruction manual provided.

Alternatively, I am trying to install GMATA in a windows computer. I hope then it works

**Comment by XuewenWangUGA** [comment]
There are details on how to complie and install the dependencies at GMATA sourceforge for Linux like and Mac OS system . After you downloaded the dependencies there, you can unzip them. There is a readme file in unzipped forder which tell you how to install.



Xuewen Wang, PhD

CAES-Applied Genetic Tech Ctr
University of Georgia

email: ***@***.***

________________________________

**Comment by ruru-adra** [comment]
Dear Wang,
I have successfully installed the primer3 and e-PCR in the Ubuntu 20 and create the link in the /usr/bin.
Both can run separately (i.e.  primer3_core   , e-PCR).
But when I run with doprimer_smat.pl , the .mk and .stats gave an empty file. (perl doprimer_smt.pl -i ~/Desktop/kr/rslt/gmata/leaf_final_assembly.fasta -sr ~/Desktop/kr/rslt/gmata/leaf_final_assembly.fasta.ssr -p u -fl 200 -l 2000 -ts F -smin 120 -Smax 400 -tm 60 -dir '/usr/bin')
I am not sure what's wrong with the installation or setting in the .pl file.
I hope you can help me to resolve this issue.

**Comment by XuewenWangUGA** [comment]
please make sure that your .fasta, .ssr is not empty. Are there any error report? Could you share the installed primer3_core, e-PCR and your input files one Google Drive so I can repeat your issue before I debug?

**Comment by ruru-adra** [comment]
Dear Wang,

This is the gdrive link: https://drive.google.com/drive/folders/1snQyzYgZW9EQZsDBQmzo7SvYDb44wSxD?usp=sharing
I also attached the verbose file derived from the doprimer.pl.
Hope it helps.

Thank you.

**Comment by XuewenWangUGA** [comment]
Could you provide your input assembly.fasta via a private link and send me email to xwang.kib@gmail.com ? You have to make sure the .fasta sequence is present.

**Comment by XuewenWangUGA** [comment]
I figured out your issue which is caused by the latest version of Primer3 has been changed. I uploaded here a version of Primer3 which is tested and compatible with GMATA V2.3 or lower version. With the tested primer3, your issue will be resolved. What needed to do is to install the tested version of Primer3. I will update code to catch up with the latest primer3 in the future.

**Comment by ruru-adra** [comment]
Dear Wang,

Thank you very much. If that's the case, I will try to re-install the one that you upload in the GitHub. I'll will let you know the update.
My net is a bit slow, hence take some time to upload the fasta file.

**Comment by XuewenWangUGA** [comment]
It is my pleasure to help you. Thanks for using GMATA.

**Comment by NHKisr** [comment]
Thanks for your response. But my issue is resolved. 
Regards,
Nazima

**Comment by XuewenWangUGA** [comment]
Great.

**Comment by bhavani-bi** [comment]
Dear wang,

I am facing issues with primer3 and epcr. Below is the error I am getting when I tried to install primer3 downloaded from the github. Please help me resolve


g++ -c -g -Wall -D__USE_FIXED_PROTOTYPES__ -O2  primer3_boulder_main.c
g++ -c -g -Wall -D__USE_FIXED_PROTOTYPES__ -O2  -o format_output.o format_output.c
g++ -c -g -Wall -D__USE_FIXED_PROTOTYPES__ -O2  -o read_boulder.o read_boulder.c
g++ -c -g -Wall -D__USE_FIXED_PROTOTYPES__ -O2  -o print_boulder.o print_boulder.c
g++ -c -g -Wall -D__USE_FIXED_PROTOTYPES__ -O2 -Wno-deprecated  -o libprimer3.o libprimer3.c
libprimer3.c: In function ‘void calc_and_check_oligo_features(const p3_global_settings*, primer_rec*, oligo_type, const dpal_arg_holder*, const thal_arg_holder*, const seq_args*, oligo_stats*, p3retval*, const char*)’:
libprimer3.c:3055:5: warning: this ‘else’ clause does not guard... [-Wmisleading-indentation]
 3055 |   } else
      |     ^~~~
libprimer3.c:3077:4: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘else’
 3077 |    if ((must_use
      |    ^~
g++ -c -g -Wall -D__USE_FIXED_PROTOTYPES__ -O2 -o p3_seq_lib.o p3_seq_lib.c
ar rv libprimer3.a libprimer3.o p3_seq_lib.o
ar: creating libprimer3.a
a - libprimer3.o
a - p3_seq_lib.o
ranlib libprimer3.a
g++ -c -g -Wall -D__USE_FIXED_PROTOTYPES__ -O2  -o dpal_primer.o dpal.c
dpal.c: In function ‘void print_align(const unsigned char*, const unsigned char*, int (*)[1600][3], int, int, const dpal_args*)’:
dpal.c:1036:5: warning: this ‘for’ clause does not guard... [-Wmisleading-indentation]
 1036 |     for(i=j;i<j+70;i++) fprintf(stderr, "%c",sy[i]); fprintf(stderr,"\n");
      |     ^~~
dpal.c:1036:54: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘for’
 1036 |     for(i=j;i<j+70;i++) fprintf(stderr, "%c",sy[i]); fprintf(stderr,"\n");
      |                                                      ^~~~~~~
ar rv libdpal.a dpal_primer.o
ar: creating libdpal.a
a - dpal_primer.o
ranlib libdpal.a
g++ -c -g -Wall -D__USE_FIXED_PROTOTYPES__ -O2 -ffloat-store  -o thal_primer.o thal.c
thal.c: In function ‘void thal(const unsigned char*, const unsigned char*, const thal_args*, thal_results*)’:
thal.c:409:16: error: ISO C++ forbids comparison between pointer and integer [-fpermissive]
  409 |    if ('\0' == oligo_f) {
      |                ^~~~~~~
thal.c:414:16: error: ISO C++ forbids comparison between pointer and integer [-fpermissive]
  414 |    if ('\0' == oligo_r) {
      |                ^~~~~~~
thal.c: In function ‘void tableStartATS(double, double (*)[5])’:
thal.c:1162:4: warning: this ‘for’ clause does not guard... [-Wmisleading-indentation]
 1162 |    for (i = 0; i < 5; ++i)
      |    ^~~
thal.c:1165:6: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘for’
 1165 |      atpS[0][3] = atpS[3][0] = atp_value;
      |      ^~~~
thal.c: In function ‘void tableStartATH(double, double (*)[5])’:
thal.c:1174:4: warning: this ‘for’ clause does not guard... [-Wmisleading-indentation]
 1174 |    for (i = 0; i < 5; ++i)
      |    ^~~
thal.c:1178:6: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘for’
 1178 |      atpH[0][3] = atpH[3][0] = atp_value;
      |      ^~~~

**Comment by jaynima** [comment]
Hi, Tried opening fasta file in GMATA (I use macOS) but in the search dialog box, I am unable to locate the fasta file

**Comment by natturne** [comment]
Hi, I have used GMATA in the past and have been able to use the GMATA.jar file just fine. Now when I am using it, I receive this output after the program says it has finished, 
"Running statement: 
 -p u -fl 400 -l 2000 -ts F -smin 120 -Smax 400 -tm 60 -dir    -i  C:\Path\to\.fasta -sr C:\Path\to\.fasta.ssr

File path:
C:\WINDOWS\system32"
When I check in this area, I fail to find any output. 

Next I have tried to fun the gmata.pl script and doprimer_smt.pl script in linux. With gmata.pl it runs fine until it gets to primer design stage, which is where it returns the output, 
"sh: primer3_core.exe: command not found
Processing primer and producing final primers.
parameters: -ts F
 Output files:
NLUmasked.fasta.ssr.mk
NLUmasked.fasta.ssr.mk.sts
NLUmasked.fasta.ssr.mk.sat3
results yielded at Fri Feb  2 19:48:11 2024

To do marker ePCR, marker allels counting.
parameters: w=12 f=1 pf=w m=3000 n=0 g=0 d=100-1000 p=-,
 Output files:
 NLUmasked.fasta.eMap
 NLUmasked.fasta.eMap.amp
 NLUmasked.fasta.eMap.frg
results yielded at Fri Feb  2 19:48:12 2024

Can't exec "e-PCR.exe": No such file or directory at elctPCR.pl line 37.
system call failed: -1 at elctPCR.pl line 37.
system perl elctPCR.pl -o eMap -pf w -w 12 -f 1 -m 3000 -d 100-1000 -n 0 -g 0 -p - -i NLUmasked.fasta -mk NLUmasked.fasta.ssr.mk.sts failed: 512 at gmata.pl line 117, <DNAhand> chunk 8."

It also says s similar thing while plotting statistical graphics saying that sh: Rscript: command not found, but the primer design is most important to me currently. I have ran the gmata.pl program multiple times using a seemingly fine installation of primer3-2.2.3 and with primer3-2.4.0, which I know you said to use the prior, but it is just not working for me. Should I look further into installation issues? Also, are there a primer3_core.sh and e-PCR.sh files available anywhere to run GMATA in linux?
